File: 30-days-of-code/cpp/day-18-queues-and-stacks.cpp
#include <iostream>
#include <queue>
#include <stack>

using namespace std;

class Solution {
    private:
        //Two instance variables: one for your stack , and one for your queue.
        stack<char> myStack;
        queue<char> myQueue;

    public:

        //A void pushCharacter(char ch) method that pushes a character onto a stack.
        void pushCharacter(char ch) {
            myStack.push(ch);
        }

        //A void enqueueCharacter(char ch) method that enqueues a character in the queue instance variable.
        void enqueueCharacter(char ch) {
            myQueue.push(ch);
        }

        //A char popCharacter() method that pops and returns the character at the top of the stack instance variable.
        char popCharacter() {
            char topChar = myStack.top();
            myStack.pop();
            return topChar;
        }

        //A char dequeueCharacter() method that dequeues and returns the first character in the queue instance variable.
        char dequeueCharacter() {
            char firstChar = myQueue.front();
            myQueue.pop();
            return firstChar;
        }

};

int main() {
    // read the string s.
    string s;
    getline(cin, s);
    
  	// create the Solution class object p.
    Solution obj;
    
    // push/enqueue all the characters of string s to stack.
    for (int i = 0; i < s.length(); i++) {
        obj.pushCharacter(s[i]);
        obj.enqueueCharacter(s[i]);
    }
    
    bool isPalindrome = true;
    
    // pop the top character from stack.
    // dequeue the first character from queue.
    // compare both the characters.
    for (int i = 0; i < s.length() / 2; i++) {
        if (obj.popCharacter() != obj.dequeueCharacter()) {
            isPalindrome = false;
            
            break;
        }
    }
    
    // finally print whether string s is palindrome or not.
    if (isPalindrome) {
        cout << "The word, " << s << ", is a palindrome.";
    } else {
        cout << "The word, " << s << ", is not a palindrome.";
    }
    
    return 0;
}

File: 30-days-of-code/php/day-18-queues-and-stacks.php
<?php

class Solution {
    public $myQueue, $myStack;

    public function __construct() {
      $this->myQueue = new SplQueue();
      $this->myStack = new SplQueue();
    }    
    //A void pushCharacter(char ch) method that pushes a character onto a stack.
    function pushCharacter($ch) {
        $this->myStack->push($ch);
    }

    //A void enqueueCharacter(char ch) method that enqueues a character in the queue instance variable.
    function enqueueCharacter($ch) {
        $this->myQueue->enqueue($ch);
    }

    //A char popCharacter() method that pops and returns the character at the top of the stack instance variable.
    function popCharacter() {
        return $this->myStack->pop();
    }

    //A char dequeueCharacter() method that dequeues and returns the first character in the queue instance variable.
    function dequeueCharacter() {
        return $this->myQueue->dequeue();
    }

}

// read the string s
$s = fgets(STDIN);

// create the Solution class object p
$obj = new Solution();

$len = strlen($s);
$isPalindrome = true;

// push/enqueue all the characters of string s to stack
for ($i = 0; $i < $len; $i++) {
    $obj->pushCharacter($s{$i});
    $obj->enqueueCharacter($s{$i});
}

/*
pop the top character from stack
dequeue the first character from queue
compare both the characters
*/
for ($i = 0; $i < $len / 2; $i++) {
    if($obj->popCharacter() != $obj->dequeueCharacter()){
        $isPalindrome = false;
    	
        break;
    }
}

//finally print whether string s is palindrome or not.
if ($isPalindrome)
    echo "The word, ".$s.", is a palindrome.";
else
    echo "The word, ".$s.", is not a palindrome.";
?>